cli/cmd/addon_download.go

```diff
@@ -74,7 +75,16 @@ func runAddonDownload(ctx context.Context, cmd *cobra.Command, args []string) er
 		return fmt.Errorf("invalid addon ref %q: version must start with 'v'", args[0])
 	}
 
+	checksum := ""
+	hc := *http.DefaultClient
+	hc.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		if len(via) == 1 && req.Response != nil {
+			checksum = req.Response.Header.Get("X-Checksum-Sha256") // get checksum from first request
```

Comment:
This is a bit weird: `via[0].Response` is supposed to be `nil` for some reason, _that_ response is available in `req.Response`